### ISSUE TITLE
bug(material): Can't find stylesheet to import(@use '@angular/cdk';)

### ISSUE THREAD

**Issue by samyak087** [angular/components#26249]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Getting below material error after migrating angular from v13 to v14.The project also consist of storybook v6.5.12.The project also consist of Angular library.

The error occurs while running the build library command for our project.

[10:37:14] erro: There is an error in your styles:
[10:37:14] erro: Can't find stylesheet to import.
[10:37:14] erro:   ╷
[10:37:14] erro: 1 │ @use '@angular/cdk';
[10:37:14] erro:   │ ^^^^^^^^^^^^^^^^^^^
[10:37:14] erro:   ╵
[10:37:14] erro:   node_modules\@angular\material\core\style\_menu-common.scss 1:1  @forward
[10:37:14] erro:   node_modules\@angular\material\_index.scss 33:1                  @use
[10:37:14] erro:   stdin 7:1                                                        root stylesheet on line (1, 1)

**Package.json**
"dependencies": {
    "@angular/animations": "^14.2.12",
    "@angular/cdk": "^14.2.7",
    "@angular/common": "^14.2.12",
    "@angular/compiler": "14.2.12",
    "@angular/core": "^14.2.12",
    "@angular/flex-layout": "^9.0.0-beta.31",
    "@angular/forms": "14.2.12",
    "@angular/localize": "^14.2.12",
    "@angular/material": "^14.2.7",
    "@angular/platform-browser": "14.2.12",
    "@angular/platform-browser-dynamic": "^14.2.12",
    "@angular/router": "^14.2.12",
    "core-js": "^3.6.5",
    "ngx-mask": "^10.0.1",
    "rxjs": "^6.6.2",
    "scss-bundle": "^3.1.2",
    "ts-jest": "^29.0.3",
    "tslib": "^2.0.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/architect": "^0.1402.10",
    "@angular-devkit/build-angular": "^14.2.10",
    "@angular-devkit/core": "^14.2.10",
    "@angular-eslint/eslint-plugin": "0.0.1-alpha.32",
    "@angular/cli": "^14.2.10",
    "@angular/compiler-cli": "14.2.12",
    "@angular/language-service": "^14.2.12",
    "@babel/core": "^7.11.1",
    "@storybook/addon-a11y": "^6.5.12",
    "@storybook/addon-actions": "^6.5.12",
    "@storybook/addon-backgrounds": "^6.5.12",
    "@storybook/addon-centered": "^5.3.21",
    "@storybook/addon-cssresources": "^6.2.9",
    "@storybook/addon-docs": "^6.5.12",
    "@storybook/addon-jest": "^6.5.12",
    "@storybook/addon-knobs": "^6.0.10",
    "@storybook/addon-links": "^6.5.12",
    "@storybook/addon-notes": "^5.3.21",
    "@storybook/addon-storysource": "^6.5.12",
    "@storybook/addon-viewport": "^6.5.12",
    "@storybook/addons": "^6.5.12",
    "@storybook/angular": "^6.5.12",
    "@storybook/builder-webpack5": "^6.5.12",
    "@storybook/manager-webpack5": "^6.5.12",
    "@storybook/theming": "^6.5.12",
    "@testing-library/angular": "^9.2.2",
    "@testing-library/jest-dom": "^5.11.3",
    "@types/jest": "^29.2.3",
    "@types/node": "^12.11.1",
    "@typescript-eslint/eslint-plugin": "^3.9.1",
    "@typescript-eslint/eslint-plugin-tslint": "^3.9.1",
    "@typescript-eslint/parser": "^3.9.1",
    "babel-loader": "^8.1.0",
    "codelyzer": "^6.0.0",
    "copyfiles": "^2.3.0",
    "eslint": "^7.7.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-jsdoc": "^28.7.0",
    "eslint-plugin-prefer-arrow": "^1.2.2",
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "jest-preset-angular": "^12.2.3",
    "jest-sonar": "^0.2.10",
    "ng-packagr": "^14.2.2",
    "protractor": "~7.0.0",
    "storybook-addon-themes": "^6.1.0",
    "storybook-dark-mode": "^1.1.2",
    "ts-node": "^8.9.0",
    "tslint": "~6.1.0",
    "typescript": "4.6.4"
  }

### Reproduction

Steps to reproduce:
1. 
2.

### Expected Behavior

Styles should compile

### Actual Behavior

Style compilation fails and throws the can't find stylesheet error.

### Environment

Angular:14.2.12
CDK/Material:14.2.7
Browser(s):Chrome
Operating System (e.g. Windows, macOS, Ubuntu):Windows

**Comment by Elvynia** [comment]
I had this error and I fixed it so I wanted to share some information. I was used to material integration code but not to the recent changes.

I got the same stacktrace until I changed my custom theme.scss code from `@include common-theme($theme);` (old way) to `@include mat.all-component-themes($theme);` (new way starting from v14 I think).
Hopefully this helps anyone landing here because of the same mistake.
